aws/resource_aws_network_acl_rule.go

```diff
@@ -139,7 +139,7 @@ func resourceAwsNetworkAclRuleCreate(d *schema.ResourceData, meta interface{}) e
 
 	// Specify additional required fields for ICMP. For the list
 	// of ICMP codes and types, see: https://www.iana.org/assignments/icmp-parameters/icmp-parameters.xhtml
-	if p == 1 {
+	if p == 1 || p == 58 {
```

Comment:
whilst this is clear in that there's a test for it - IMO we want a comment/constant to explain what these are so it doesn't require a lookup

Reply:
I hope https://github.com/terraform-providers/terraform-provider-aws/issues/6302 covers this here as well. Probably best to solve this "globally". 😄